content/en/blog/_posts/2024-02-22-diy-create-your-own-cloud-with-kubernetes-part-2/index.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: blog
+title: "DIY: Create Your Own Cloud with Kubernetes (Part 2)"
+slug: diy-create-your-own-cloud-with-kubernetes-part-2
+date: 2024-02-04
+---
+
+**Author**: Andrei Kvapil (Ænix)
+
+Continuing our series of posts on how to build your own cloud using just the Kubernetes ecosystem.
+In the [previous article](/blog/2024/01/22/diy-create-your-own-cloud-with-kubernetes-part-1/), we explained how we prepare a basic Kubernetes distribution based on Talos Linux and Flux CD.
+In this article, we'll show you a few various virtualization technologies in Kubernetes and prepare everything need to run virtual machines in Kubernetes, primarily storage and networking.
+
+We will talk about technologies such as KubeVirt, LINSTOR, and Kube-OVN.
+
+But first, let's explain why virtual machines are needed for, and why can't you just use docker containers for building cloud?
+The reason is that containers do not provide a sufficient level of isolation.
+Although the situation improves year by year, we often encounter vulnerabilities that allow escaping the container sandbox and elevating privileges in the system.
+
+On the other hand, Kubernetes was not originally designed to be multi-tenant system, meaning the basic usage pattern involves creating a separate Kubernetes cluster for every independent project and development team.
+
+Virtual machines are the primary means of isolating tenants from each other in a cloud. In virtual machines, users are allowed to run any code with administrative rights, but this don't affect other tenants or the cloud system itself. In other words, virtual machines allows to achieve [hard multi-tenancy isolation](https://kubernetes.io/docs/concepts/security/multi-tenancy/#isolation), and run in environments where tenants do not trust each other.
+
+### Virtualization technologies in Kubernetes
+
+There are several different technologies that bring virtualization into the Kubernetes world: [KubeVirt](https://kubevirt.io/) and [Kata Containers](https://katacontainers.io/)
+are the most popular ones. But you should know that they work differently.
+
+**Kata Containers** implements the CRI (Container Runtime Interface) and provides an additional level of isolation for standard containers by running them in virtual machines. But they work in a same single Kubernetes-cluster.
+
+![Simplified diagram of Kata-Containers](kata-containers.svg)
```

Comment:
Consider using [figure](https://gohugo.io/content-management/shortcodes/#figure) shortcode to display images in the blog content.